src/memoization.ml

```diff
@@ -0,0 +1,311 @@
+open !Stdune
+open Import
+(* open Staged *)
+open Fiber.O
+
+type name = string
+type ser_input = string
```

Comment:
Is it worth changing the serialization format to s expressions before we integrate things? I think that should make it easier to have input specs that handle pairs / lists of inputs etc.

Reply:
In practice, we will probably just use marshal as it's faster.